src/main/webapp/app/exam/manage/student-exams/student-exam-detail-table-row/student-exam-detail-table-row.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, OnChanges, Input } from '@angular/core';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { Submission } from 'app/entities/submission.model';
+import { StudentParticipation } from 'app/entities/participation/student-participation.model';
+import { getExerciseSubmissionsLink, getLinkToSubmissionAssessment } from 'app/utils/navigation.utils';
+import { round } from 'app/shared/util/utils';
+import { Course } from 'app/entities/course.model';
+import { Result } from 'app/entities/result.model';
+
+@Component({
+    /* tslint:disable-next-line component-selector */
+    selector: '[jhi-student-exam-detail-table-row]',
+    templateUrl: './student-exam-detail-table-row.component.html',
+    providers: [],
+})
+export class StudentExamDetailTableRowComponent implements OnChanges {
+    @Input() exercise: Exercise;
+    @Input() examId: number;
+    @Input() isTestRun: boolean;
+    @Input() course: Course;
+    @Input() busy: boolean;
+
+    courseId: number;
+    studentParticipation: StudentParticipation;
+    submission: Submission;
+    result: Result;
+    openingAssessmentEditorForNewSubmission = false;
+    readonly ExerciseType = ExerciseType;
+
+    ngOnChanges() {
+        if (this.exercise.studentParticipations?.[0]) {
+            this.studentParticipation = this.exercise.studentParticipations![0];
+            if (this.studentParticipation.submissions?.length! > 0) {
+                this.submission = this.studentParticipation.submissions![0];
+            }
+            if (this.studentParticipation.results?.length! > 0) {
+                this.result = this.studentParticipation.results![0];
+            }
+        }
+        if (this.course && this.course.id) {
+            this.courseId = this.course.id!;
+        }
+    }
+
+    /**
+     * Get an icon for the type of the given exercise.
+     * @param exercise {Exercise}
+     */
+    exerciseIcon(exercise: Exercise): string {
```

Comment:
We should think about re-using the method of the exercise model:
https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/entities/exercise.model.ts#L146-L158

Reply:
done